src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerTypeSymbol.cs

```diff
@@ -0,0 +1,136 @@
+
+        private FunctionPointerTypeSymbol(FunctionPointerMethodSymbol signature)
+        {
+            _signature = signature;
+        }
+
+        private FunctionPointerMethodSymbol _signature;
+        public MethodSymbol Signature => _signature;
+
+        public override bool IsReferenceType => false;
+        public override bool IsValueType => true;
+        public override TypeKind TypeKind => TypeKind.FunctionPointer;
+        public override bool IsRefLikeType => false;
+        public override bool IsReadOnly => false;
+        public override SymbolKind Kind => SymbolKind.FunctionPointer;
+        public override Symbol? ContainingSymbol => null;
+        public override ImmutableArray<Location> Locations => ImmutableArray<Location>.Empty;
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => ImmutableArray<SyntaxReference>.Empty;
+        public override Accessibility DeclaredAccessibility => Accessibility.NotApplicable;
+        public override bool IsStatic => false;
+        public override bool IsAbstract => false;
+        public override bool IsSealed => false;
+        // Pointers do not support boxing, so they really have no base type.
+        internal override NamedTypeSymbol? BaseTypeNoUseSiteDiagnostics => null;
+        internal override ManagedKind ManagedKind => ManagedKind.Unmanaged;
+        internal override ObsoleteAttributeData? ObsoleteAttributeData => null;
+        public override void Accept(CSharpSymbolVisitor visitor) => visitor.VisitFunctionPointerType(this);
+        public override TResult Accept<TResult>(CSharpSymbolVisitor<TResult> visitor) => visitor.VisitFunctionPointerType(this);
+        public override ImmutableArray<Symbol> GetMembers() => ImmutableArray<Symbol>.Empty;
+        public override ImmutableArray<Symbol> GetMembers(string name) => ImmutableArray<Symbol>.Empty;
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers() => ImmutableArray<NamedTypeSymbol>.Empty;
+        public override ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name) => ImmutableArray<NamedTypeSymbol>.Empty;
+        internal override TResult Accept<TArgument, TResult>(CSharpSymbolVisitor<TArgument, TResult> visitor, TArgument a) => visitor.VisitFunctionPointerType(this, a);
+        internal override ImmutableArray<NamedTypeSymbol> InterfacesNoUseSiteDiagnostics(ConsList<TypeSymbol>? basesBeingResolved = null) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override bool Equals(TypeSymbol t2, TypeCompareKind compareKind, IReadOnlyDictionary<TypeParameterSymbol, bool>? isValueTypeOverrideOpt = null)
+        {
+            if (!(t2 is FunctionPointerTypeSymbol other))
+            {
+                return false;
+            }
+
+            return _signature.Equals(other._signature, compareKind, isValueTypeOverrideOpt);
+        }
+
+        public override int GetHashCode()
+        {
+            return 19 * _signature.GetHashCode();
```

Comment:
>19 [](start = 19, length = 2)

We usually do not use magic numbers to adjust hash codes. Do you expect the pointer and the underlying function be placed in the same dictionary? If it is really important to have distinct codes, could we use one of the functions we use to combine hash codes, or add one specifically for cases like this?

Reply:
> Do you expect the pointer and the underlying function be placed in the same dictionary?

They are both symbols, I wouldn't consider it out of the realm of possibility. I'll do something in our utils.

---
In reply to: [362668456](https://github.com/dotnet/roslyn/pull/40431#discussion_r362668456) [](ancestors = 362668456)

Reply:
> They are both symbols, I wouldn't consider it out of the realm of possibility. I'll do something in our utils.

Consider that hash code doesn't have to be unique (even unique ones often still end up in the same bucket), it just have to bee good enough to avoid major collisions. When two symbols collide, it is really not a problem I think. Especially, that I would find it extremely rare when someone would want to put both the pointer and its signature in the same dictionary.

---
In reply to: [362679025](https://github.com/dotnet/roslyn/pull/40431#discussion_r362679025) [](ancestors = 362679025,362668456)